### DISCUSSION TITLE
RandomActivation Scheduler question

### DISCUSSION THREAD

**Discussion by mauruscz** [mesa/mesa#2106] [Q&A]
Hi there,
I have a small doubt regarding the RandomActivation Scheduler, so I will make an example for the Schelling model.

Say that two agents, _A_ and _B_, at the step _k_ are unhappy and decide to move.
_A_ is randomly woken up before _B_ and decide to go in the cell _C_
Then _B_ is woken up and, given that they evaluated the situation of the grid in the same step, it randomly decide to go to cell _C_ too.

What happens to _B_? Does the scheduler automatically look for another cell or does it stay in its unhappy place for the step and wait for the next step?

Somehow I understood that the correct answer is the second, but I'd like to be 100% sure, and maybe having a reference.

Thank you so much.

**Comment by quaquel** [comment]
If B is woken up second, cell C is no longer empty. So B will go to another random empty cell. Looking for a random empty cell is not the responsibility of the scheduler by the way.

**Reply by mauruscz** [reply]
Of course is not responsibility of the Scheduler, I was making the example of the Schelling model :)
